lib/Driver/Driver.cpp

```diff
@@ -199,6 +200,18 @@ static void validateArgs(DiagnosticEngine &diags, const ArgList &Args) {
       diags.diagnose(SourceLoc(), diag::invalid_conditional_compilation_flag,
                      name);
   }
+
+  if (auto *forceLoadArg = Args.getLastArg(options::OPT_autolink_force_load)) {
+    if (auto *incrementalArg = Args.getLastArg(options::OPT_incremental)) {
+      // Note: -incremental can itself be overridden by other arguments later
+      // on, but since -autolink-force-load is a rare and not-really-recommended
+      // option it's not worth modeling that complexity here (or moving the
+      // check somewhere else).
+      diags.diagnose(SourceLoc(), diag::error_option_not_supported,
+                     forceLoadArg->getSpelling(),
+                     incrementalArg->getSpelling());
+    }
+  }
```

Comment:
I would suggest pulling this block (204-214) out into a separate (static) function. I know it doesn't do much, just diagnoses a problem. But as is, it lengthens an already-too-long-for-quick-comprehension function. Ideally, every such step in this validate function would be it's own subroutine, making it possible to read it as a series of steps (validateThis(); validateThat();) at a glance. IMO, such a transformation would be worthwhile in terms of effort consumed vs future benefit to new maintainers.

Reply:
I still don't see this. *Why* is it better to have five separate validations broken up into separate functions? They are not reusable components and are called in exactly one place.